test/run-drun/pass-class-self.mo

```diff
@@ -0,0 +1,13 @@
+import { debugPrint; principalOfActor } = "mo:⛔";
+
+actor class C() = Self {
+    public func method() : async () { debugPrint "YESS!"; ignore principalOfActor Self };
+    func caller(_callee : shared () -> async ()) { };
+
+    debugPrint "Before!";
+    ignore principalOfActor Self;
+    caller(Self.method);
+    caller(method);
+    //debugPrint (debug_show(principalOfActor Self));
```

Comment:
Bizarre: uncommenting this gives a forward declaration inference error... but we already use `debugPrint` above!

Reply:
@crusso any idea how to explain this?

Reply:
Is this resolved?

Reply:
I just played around with it. I expect it's because the call to debugPrint is the last declaration that is caussing the issue...

This also produces the error:
```
import { debugPrint; principalOfActor } = "mo:⛔";

actor class C() = Self {
    debugPrint "So far so good!" 
}
```

Reply:
Filed an issue 
https://github.com/dfinity/motoko/issues/4733

Reply:
I suspected that it is unrelated. Thanks for checking!